chirp/drivers/retevis_ra25.py

```diff
@@ -0,0 +1,1058 @@
+    [(476425000, 477412501)]
+]
+
+MEMORY_REGIONS = [
+    (0x0022000, 2, 32, 'W', 'chans_present', 'Present Chan.'),
+    (0x0032000, 2, 32, 'W', 'chans_scanned', 'Scanned Chan.'),
+    (0x0042000, 200, 32, 'W', 'memory', 'Memory Chan. 1-200'),
+    (0x0052000, 200, 32, 'S', 'memory2', 'Memory Chan. 201-400'),
+    (0x0062000, 100, 32, 'S', 'memory3', 'Memory Chan. 401-500'),
+    (0x1002000, 1, 32, 'W', 'func_settings', 'Function Menu'),
+    (0x2001000, 16, 16, 'W', 'dtmf_list', 'DTMF Encode List'),
+    (0x2011000, 1, 16, 'W', 'dtmf_menu', 'DTMF Menu'),
+    (0x2031000, 1, 16, 'W', 'dtmf_start', 'DTMF PTT ID Start'),
+    (0x2041000, 1, 16, 'W', 'dtmf_end', 'DTMF PTT ID End'),
+    (0x2071000, 1, 16, 'W', 'remote_stun', 'DTMF Remote Stun'),
+    (0x2081000, 1, 16, 'W', 'remote_kill', 'DTMF Remote Kill'),
+    (0x3011000, 1, 16, 'W', 'scan_menu', 'Scan Menu'),
+    (0x3021000, 1, 16, 'W', 'startup_display', 'Startup Display'),
+    (0x8002000, 1, 32, 'RO', 'immut_1', 'Serial Number'),
+    (0x8002001, 1, 32, 'RO', 'immut_2', 'Production Date')
+]
+
+
+def send_cmd(serial, cmd, length):
+    try:
+        resp = b''
+        serial.write(cmd)
+        serial.flush()
+        resp = serial.read(length)
+        if len(resp) != length:
+            err = (f'Data send expected {length} got {len(resp)}')
+            log.error(err)
+            raise errors.RadioError(err)
+    except Exception as e:
+        raise errors.RadioError(f'Error sending to serial {e}')
+    return resp
+
+
+def start_programming(serial):
+    try:
+        r = send_cmd(serial, b'PROGRAM', 1)
+        if r != b'\x06':
+            err = (f'Enter Programming failed '
+                   f'expected 0x06 got {hex(r[0])}')
+            log.error(err)
+            raise errors.RadioError(err)
+    except Exception as e:
+        raise errors.RadioError(f'Error Entering Program Mode {e}')
+    return True
+
+
+def end_programming(serial):
+    try:
+        r = send_cmd(serial, b'END', 1)
+        if r != b'\x06':
+            log.error(f'Error {hex(r[0])}')
+    except Exception as e:
+        raise errors.RadioError(f"Error {e}")
+
+
+def parse_response(r):
+    ck = sum(r[:-1]) % 256
+    if ck != r[-1]:
+        err = f'Data check sum failed calc {ck} got {r[-1]}'
+        log.error(err)
+    return r[5:-1]
+
+
+def get_ident(serial):
+    try:
+        bs = send_cmd(serial, b'\x02', 16)
+        model, mode, ver = struct.unpack('>x7sB4s3x', bs)
+        log.info(f'Model {model} Ver {ver} Band Mode {mode}')
+        if ver in SUPPORTED_MODEL_VERSIONS[model]:
+            log.info(f'Model {model} Ver: {ver} is supported')
+        else:
+            err = f'ERROR: Model: {model} Ver: {ver} is not supported!'
+            log.error(err)
+            raise errors.RadioError(err)
+        return bs
+    except Exception as e:
+        raise errors.RadioError(e)
+
+
+def get_bands(mode):
+    # Radio skips 0xA-0XF, 0x10 == 10 :/
+    try:
+        idx = int(hex(mode)[2:]) - 1
```

Comment:
This still makes no sense to me:
```
>>> int(hex(2)[2:])
2
```
Why are you converting to hex, snipping off the `0x` and re-parsing this as a base-ten number (which can break) and then indexing it as such in your own array? Just index your array as  the proper number? And if this ever happens, this will break:
```
>>> int(hex(12)[2:])
Traceback (most recent call last):
  File "<python-input-5>", line 1, in <module>
    int(hex(12)[2:])
    ~~~^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'c'
```